src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -1854,6 +1854,26 @@ public bool IsDevelopmentMode()
         {
             return _runtimeConfigProvider.GetConfig().IsDevelopmentMode();
         }
+
+        public ForeignKeyDefinition GetFKDefinition(string sourceEntityName, string targetEntityName, string referencedEntityName, string referencingEntityName)
+        {
+            if (GetEntityNamesAndDbObjects().TryGetValue(sourceEntityName, out DatabaseObject? sourceDbObject) &&
```

Comment:
aren't sourceEntityName and referencingEntityName the same entity?

Reply:
No. Its not always the case. Source entity/target entity depend on the config how they are define. 
Eg. For a relationship between Book-Publisher defined in the Publisher's configuration:

source entity: Publisher
target entity: Book
Referencing entity: Book
Referenced entity: Publisher